packages/legend-application-query/src/components/QueryEditor.tsx

```diff
@@ -82,54 +87,119 @@ import {
   QueryBuilderNavigationBlocker,
   type QueryBuilderState,
 } from '@finos/legend-query-builder';
+import { LegendQueryTelemetryHelper } from '../application/LegendQueryTelemetryHelper.js';
 
 const QueryExportDialogContent = observer(
   (props: { exportState: QueryExportState }) => {
     const { exportState } = props;
     const applicationStore = useApplicationStore();
-    const allowCreate = exportState.allowPersist;
+
+    const [isExistingQueryName, setIsExistingQueryName] = useState(false);
+
+    const allowSaveNewQuery = exportState.allowPersist;
     const allowSave = exportState.allowPersist && exportState.allowUpdate;
```

Comment:
just to be consistent as `getExportConfigurationDirectSave`, maybe `allowDirectSave`?